### PR TITLE
Updated go version to 1.16

### PR DESCRIPTION
**Pull request by venkataramant**
1. Updated go version to 1.16
2. Makefile is updated "go mod vendor"

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,6 +1,6 @@
 NAME = heckler
 BUILD_GIT_SHA = $(shell git rev-parse --short HEAD)
-IMAGE = dockerhub.com/lollipopman/$(NAME)
+IMAGE = dockerhub.com/heckler/$(NAME)
```
Comment (ssgelm) [inline Makefile:3]: Does this actually work? I don't believe there is currently a heckler user for dockerhub, and I can't seem to find the image referenced.

File: Makefile [hunk 2]
```diff
@@ -10,6 +10,7 @@ export CC := /usr/local/musl/bin/musl-gcc
 GO_LDFLAGS := -X main.Version=$(HECKLER_VERSION) -extldflags=-static -linkmode=external
 export GOFLAGS := -mod=vendor -tags=static,osusergo
 export GOCACHE := $(CURDIR)/.go-build
+export GO111MODULE := on
```
Comment (ssgelm) [inline Makefile:13]: Is this necessary? As far as I can tell in go1.16 it defaults to on already.

**Review by ssgelm — CHANGES_REQUESTED** [review]
This looks pretty good as a whole. Could you clarify the two questions I asked, and also explain what the Makefile changes you made here do?

File: Makefile (hunk 1 above)
Comment (getaaron) [reply Makefile:3]: I was just wondering the same thing.

Comment (venkataramant) [reply Makefile:3]: We are not pushing any images to the public docker hub. For local testing and builds we can have any "org" name.
I validated all make commands , able to build binaries with this change

File: Makefile (hunk 2 above)
Comment (venkataramant) [reply Makefile:13]: "go mod vendor" is expecting this "GO111MODULE" to be ON. This is for both local/mac and docker env to run make commands

File: go.mod [hunk 3]
```diff
@@ -31,3 +31,5 @@ require (
 	google.golang.org/grpc v1.28.1
 	gopkg.in/yaml.v3 v3.0.0-20200313102051-9f266ea9e77c
 )
+
+replace github.com/braintree/heckler => ../heckler
```
Comment (lollipopman) [inline go.mod:35]: did you mean to commit this change or was this just for testing?

File: Makefile (hunk 2 above)
Comment (lollipopman) [reply Makefile:13]: I think this was only true for go 1.15, see https://go.dev/ref/mod#mod-commands for more details

File: Makefile (hunk 1 above)
Comment (lollipopman) [reply Makefile:3]: perhaps add a comment indicating that the org name doesn't matter for localdev?

File: Makefile [hunk 4]
```diff
@@ -64,18 +65,19 @@ build: vendor/github.com/libgit2/git2go/v31/static-build ## Build heckler, usual
 	go build -o . -ldflags '$(GO_LDFLAGS)' ./...
 
 .PHONY: vet
-vet: ## Vet heckler, usually called inside the container
+vet: vendor/github.com/libgit2/git2go/v31/static-build ## Vet heckler, usually called inside the container
```
Comment (lollipopman) [inline Makefile:68]: Should this change and the change to the deb target be part of a separate commit, since they are not related to go.mod 1.16?

**Review by lollipopman — COMMENTED** [review]
Thanks for this pull request @venkataramant. I left a few comments. Also, since heckler was already being built with 1.16, https://github.com/braintree/heckler/blob/main/Dockerfile#L6, perhaps the commit message should reflect that this commit upgrades the go.mod file to 1.16?

File: go.mod (hunk 3 above)
Comment (venkataramant) [reply go.mod:35]: "replace github.com/braintree/heckler => ../heckler"  for this commit I will remove it. We are adding a Change Management feature(For SNOW/JIRA/3rd party Tickets systems) in the next PRs. I will add it later.

File: Makefile (hunk 1 above)
Comment (venkataramant) [reply Makefile:3]: yes

**Review by ssgelm — APPROVED** [review]
